File: frontend/src/types/prediction.ts
export interface AlternativePrediction {
  plant: string;
  disease: string;
  confidence: number;
  raw_label: string;
}

export interface PredictionResult {
  plant: string;
  disease: string;
  confidence: number;
  confidence_level: 'High' | 'Moderate' | 'Low';
  raw_label: string;
}

export interface DiseaseReport {
  description: string;
  symptoms: string[];
  management: string[];
  prevention: string[];
  severity: string;
  disclaimer: string;
}

export interface PredictResponse {
  success: boolean;
  prediction: PredictionResult;
  alternatives: AlternativePrediction[];
  report: DiseaseReport;
  image_url?: string;
  warning?: string;
}

export interface HistoryItem {
  id: number;
  session_id: string;
  image_filename?: string;
  plant: string;
  disease: string;
  confidence: number;
  confidence_level: string;
  model_id: string;
  created_at: string;
}

export interface HistoryResponse {
  success: boolean;
  total: number;
  items: HistoryItem[];
}

export interface ModelInfo {
  model_id: string;
  model_loaded: boolean;
  classes_count: number;
  is_mock: boolean;
}

File: backend/app/schemas/prediction.py
from typing import List, Optional
from pydantic import BaseModel, Field


class AlternativePrediction(BaseModel):
    plant: str
    disease: str
    confidence: float
    raw_label: str


class PredictionResult(BaseModel):
    plant: str
    disease: str
    confidence: float
    confidence_level: str = Field(..., description="High, Moderate, or Low")
    raw_label: str


class DiseaseReport(BaseModel):
    description: str
    symptoms: List[str]
    management: List[str]
    prevention: List[str]
    severity: str = "Moderate"
    disclaimer: str = (
        "AI-generated plant analysis may be incorrect. For valuable crops or serious outbreaks, "
        "consider consulting a qualified agricultural professional."
    )


class PredictResponse(BaseModel):
    success: bool
    prediction: PredictionResult
    alternatives: List[AlternativePrediction]
    report: DiseaseReport
    image_url: Optional[str] = None
    warning: Optional[str] = None
